TheLastScrub/delta-green-foundry-vtt-system-unofficial
# Linting and Formatting

This PR introduces ESLint and Prettier to help keep the code consistent between contributors, and to reduce syntax errors potentially leading to bugs.

NOTE: This PR does two other things which helped fix some linting errors: 

1. Moves the `Handlebars.registerHelper(...)` calls to their own file (`module/other/register-helpers.js`).
2. Moves the Macro functions to their own files (`module/other/macro-functions.js`)

Steps to test:

NOTE: If you wish to review the code changes in this, I would look at the diffs for each commit,  rather than all at once

1. Open VS Code
3. Check out this branch
4. Download the ESLint extensions and Prettier extensions
5. Open a terminal and run `npm i`
6. You should see a new `/node_modules` folder that is ignored by git.
7. Open a javascript like `actor-sheet.js` and notice that ESLint is highlighting warnings in yellow.
8. Try changing some of the syntax (removing semi-colons, adding multiple blank lines between functions, removing spaces between function names and the opening brackets, etc.)
9. Notice that ESLint gives red errors and that saving auto-fixes these errors
10. IMPORTANT LAST STEP: Test as much as you can in the system to make sure that reformatting didn't break anything. Check that sheets look right, buttons behave correctly, etc. Formatting shouldn't break anything, but fixing some of the linting errors has the potential to.
